### PR TITLE
Add optimistic votes on the post page

### PR DESCRIPTION
**Pull request by funnyboy-roks**

### CODE REVIEW

File: src/routes/(app)/post/[post_id]/+page.svelte [hunk 1]
```diff
@@ -22,26 +22,37 @@
 
     export let data;
     let user_vote: 'UP' | 'DOWN' | null = data.user_vote;
```
Comment (jackjohn7) [inline src/routes/(app)/post/[post_id]/+page.svelte:24]: nit: should be camelCase

File: src/routes/(app)/post/[post_id]/+page.svelte [hunk 2]
```diff
@@ -22,26 +22,37 @@
...
+    let upvotes = data.post.upvotes;
+    let downvotes = data.post.downvotes;
```
Comment (jackjohn7) [inline src/routes/(app)/post/[post_id]/+page.svelte:26]: nit: Could be
```typescript
let { upvotes, downvotes } = data.post;
```
A bit prettier and more cleanly expandable.

File: src/hooks.server.ts [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
 import { updateLocals } from '$lib/server/auth';
+import { sleep } from '$lib/util';
 import type { Handle } from '@sveltejs/kit';
+import { env } from '$env/dynamic/private';
 
 export const handle: Handle = async ({ event, resolve }) => {
+    // Add artificial delay when in dev so we can see what the web requests will be like
+    if (dev) {
+        await sleep(+(env.DELAY ?? 500));
+    }
```
Comment (jackjohn7) [inline src/hooks.server.ts:11]: Not sure if I like this 500 ms as default in dev. This is lower than prod on my machine lol.

This does do a great job of illustrating how the optimistic actions work, but it's not super nice for just navigating the site.

**Review by jackjohn7 — CHANGES_REQUESTED** [review]
Nicely done. Needs conversation or changes

File: src/hooks.server.ts (hunk 3 above)
Comment (funnyboy-roks) [reply src/hooks.server.ts:11]: Yeah, that's fair.  I'll make it 0 by default.

File: src/routes/(app)/post/[post_id]/+page.svelte (hunk 2 above)
Comment (funnyboy-roks) [reply src/routes/(app)/post/[post_id]/+page.svelte:26]: I'm disappointed in myself as a JS developer.

File: src/routes/(app)/post/[post_id]/+page.svelte (hunk 1 above)
Comment (funnyboy-roks) [reply src/routes/(app)/post/[post_id]/+page.svelte]: :disappointed:

**Review by jackjohn7 — APPROVED** [review]

**Pull request merged**